src/mlpack/methods/ann/rbm/spike_slab_rbm_policy.hpp

```diff
@@ -0,0 +1,223 @@
+  const DataType& Parameters() const { return parameter; }
+  //! Modify the initial point for the optimization.
+  DataType& Parameters() { return parameter; }
+
+  //! Get the weight variables
+  arma::cube const& Weight() const { return weight; }
+  arma::cube& Weight() { return weight; }
+
+  //! Get the regulaliser associated with spike variables
+  DataType const& SpikeBias() const { return spikeBias; }
+  DataType& SpikeBias() { return spikeBias; }
+
+  //! Get the regulaliser associated with slab variables
+  ElemType const& SlabPenalty() const { return invSlabPenalty; }
+
+  //! Get the regulaliser associated with visible variables
+  DataType const& VisiblePenalty() const { return visiblePenalty; }
+  DataType& VisiblePenalty() { return visiblePenalty; }
+
+  //! Get the visible size
+  size_t const& VisibleSize() const { return visibleSize; }
+  //! Get the hidden size
+  size_t const& HiddenSize() const { return hiddenSize; }
+  //! Get the pool size
+  size_t const& PoolSize() const { return poolSize; }
+
+ private:
+  /**
+   * Spike Mean function calculates the following distribution
+   * P(h|v) which is given by
+   *  sigm(v^T*W_i*\alpha_i^{-1}*W_i^T*v + b_i)
+   *
+   * @param visible the visible layer
+   * @param spikeMean hidden layer
+   */
+  void SpikeMean(DataType&& visible, DataType&& spikeMean);
+  /**
+   * Sample Spike function samples the spike
+   * function using bernoulli distribution
+   * @param spikeMean indicates P(h|v)
+   * @param spike the sampled 0/1 spike variables
+   */
+  void SampleSpike(DataType&& spikeMean, DataType&& spike);
+
+  /**
+   * SlabMean function calculates the mean of 
+   * normal distribution of P(s|v,h).
+   * Where the mean is given by h_i*\alpha^{-1}*W_i^T*v 
+   * 
+   * @param visible the visible layer units
+   * @param spike the spike variables from hidden layer
+   * @param slabMean the mean of the normal distribution
+   */
+  void SlabMean(DataType&& visible, DataType&& spike,
+      DataType&& slabMean);
+  /**
+   * SampleSlab function calculates the
+   * normal distribution P(s|v,h).
+   * Where the mean is given by h_i*\alpha^{-1}*W_i^T*v 
+   * variance is givenby \alpha^{-1}
+   *
+   * @slabMean mean of the normal distribution
+   * @slab sample slab variable from the normal distribution
+   */
+  void SampleSlab(DataType&& slabMean, DataType&& slab);
+
+  //! Locally stored parameters number of visible neurons
+  size_t visibleSize;
+  //! Locally stored parameters number of hidden neurons
+  size_t hiddenSize;
+  //! Locally stored parameters poolSize 
+  size_t poolSize;
+  //! Locally stored parameters
+  DataType parameter;
+  //! Locally stored weight of the network (visibleSize * poolSize * hiddenSize)
+  arma::cube weight;
+  //! Locally stored spikeBias (hiddenSize * 1)
+  DataType spikeBias;
+  //! Locally stored slabPenalty
+  ElemType slabPenalty;
+  //! Locally stored iverse of slabPenalty
+  ElemType invSlabPenalty;
```

Comment:
It seems you did an optimization here. But I think the compiler is clever enough, so we can remove `invSlabPenalty` at all in order to improve readability.